tests/integration/server/proposal_api_tests.py

```diff
@@ -0,0 +1,120 @@
+        }
+        manager3_create_response = session.post(
+            "http://rbac-server:8000/api/users", json=manager3_create_payload
+        )
+        manager3_id = manager3_create_response.json()["data"]["user"]["id"]
+        manager4_create_payload = {
+            "name": "manager4",
+            "username": "manager4_username",
+            "password": "manager4_password",
+            "email": "manager4@email_id",
+            "manager": manager3_id,
+        }
+        manager4_create_response = session.post(
+            "http://rbac-server:8000/api/users", json=manager4_create_payload
+        )
+        manager4_id = manager4_create_response.json()["data"]["user"]["id"]
+        manager5_create_payload = {
+            "name": "manager5",
+            "username": "manager5_username",
+            "password": "manager5_password",
+            "email": "manager5@email_id",
+            "manager": manager4_id,
+        }
+        manager5_create_response = session.post(
+            "http://rbac-server:8000/api/users", json=manager5_create_payload
+        )
+        manager5_id = manager5_create_response.json()["data"]["user"]["id"]
+        roleowner_creation_payload = {
+            "name": "roleowner_name",
+            "username": "roleowner_username",
+            "password": "roleowner_password",
+            "email": "roleowner_name@email_id",
+            "manager": manager5_id,
+        }
+        roleowner_creation_response = session.post(
+            "http://rbac-server:8000/api/users", json=roleowner_creation_payload
+        )
+        roleowner_id = roleowner_creation_response.json()["data"]["user"]["id"]
+        create_role_payload = {
+            "name": "Test_role",
+            "owners": roleowner_id,
+            "administrators": roleowner_id,
+        }
+        role_creation_response = session.post(
+            "http://rbac-server:8000/api/roles", json=create_role_payload
+        )
+        role_id = role_creation_response.json()["data"]["id"]
+        creating_user_payload = {
+            "name": "user_name",
+            "username": "user_username",
+            "password": "user_password",
+            "email": "user@email_id",
+        }
+        creating_user_response = session.post(
+            "http://rbac-server:8000/api/users", json=creating_user_payload
+        )
+        user_id = creating_user_response.json()["data"]["user"]["id"]
+        creating_proposal_payload = {"id": user_id, "reason": "add the test member"}
+        creating_proposal_response = session.post(
+            "http://rbac-server:8000/api/roles/" + role_id + "/members",
+            json=creating_proposal_payload,
+        )
+        proposal_id = creating_proposal_response.json()["proposal_id"]
+        proposal_information = session.get(
+            "http://rbac-server:8000/api/proposals/" + proposal_id
+        )
+        assert roleowner_id in proposal_information.json()["data"]["approvers"]
```

Comment:
each of these tests/assertions should be their own function, you can look at some of the other integration tests to view examples of pytest parameters and fixtures which might help. see /tests/integration/providers/ldap/delta_inbound_sync_tests.py, 

any resources you create during your tests need to be deleted on completion as well to prevent side effects. Look in delta_inbound_sync_tests.py for examples of using pytest teardown functions to do this.

Reply:
these users are created using api ,s and these can only be deleted using delete user by api  which is currently in progress for this sprint one it competes we can delete them

Reply:
Can you add TODOs and create a ticket (if you haven't already) for that and link it to this PR and the ticket that provides a user deletion api?

Reply:
I updated it please check it

Reply:
1. The doc string for this test should be more clear... The test creates a hierarchy of users, a role, one more additional test user and then a proposal to add that last test user to the role.  The goal being to test that all of the users in the management chain will be approvers for the proposal.
2. Creating all these users individually is excessively verbose. Maybe make a list of dictionaries with their details, and  iterate over that to create them instead?